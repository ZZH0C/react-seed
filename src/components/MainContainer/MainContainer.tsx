import React from 'react';
type MainContainerProps = React.HTMLAttributes<HTMLElement>;

export const MainContainer: React.FC<MainContainerProps> = ({ children }) => {
  return <main className="container main">{children}</main>;
};
