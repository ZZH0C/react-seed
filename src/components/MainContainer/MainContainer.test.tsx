import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { MainContainer } from './MainContainer';

describe('components/ButtonLink', () => {
  const mockProps = {
    children: 'some text',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(<MainContainer>{mockProps.children}</MainContainer>),
    ).toMatchSnapshot();
  });
  it('should render without children and match snapshot ', () => {
    expect(shallow(<MainContainer />)).toMatchSnapshot();
  });
});
