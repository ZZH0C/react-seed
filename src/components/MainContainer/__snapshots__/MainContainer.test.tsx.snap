// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<main
  className="container main"
>
  some text
</main>
`;

exports[`components/ButtonLink should render without children and match snapshot  1`] = `
<main
  className="container main"
/>
`;
